storm-core/pom.xml

```diff
@@ -602,6 +612,42 @@
                           <pattern>org.eclipse.jetty</pattern>
                           <shadedPattern>org.apache.storm.shade.org.eclipse.jetty</shadedPattern>
                         </relocation>
+                        <relocation>
+                            <pattern>com.codahale.metrics</pattern>
+                            <shadedPattern>org.apache.storm.shade.com.codahale.metrics</shadedPattern>
+                        </relocation>
```

Comment:
Will include a comment in the documentation to state that reporting via JMX is supported at this time since shading may prevent other plugins from working correctly.  Will file a follow-up jira to investigate whether supporting additional plugins is need